### PR TITLE
chore: Transform extension optimization.

### PR DESCRIPTION
**Pull request by roman-effrimov**
## Purpose of this PR
* Solved wrong test messages from previous task(`RectTransformExtenstionsTests`)
* Performance optimization on `transform.Clear() `extension method. 
* Tiny code optimization inside `transform.FindOrCreateChild()` method
* All changes based on the next [review](https://github.com/StansAssets/com.stansassets.foundation/issues/101). 

## Testing status

* Added unit tests for `transform.Clear()` and `transform.FindOrCreateChild()` methods. 
* Performed performance tests with deleting a 100 000 empty `GameObjects` with` transform.Clear()` method. The new code deleted 100 000 objects 5 times faster than previous version of the mehod. 
### Manual testing status
Manually tested all methods to work the same as previous versions inside the Editor.

### CODE REVIEW

File: Tests/Editor/Extensions/TransformExtenstionsTest.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
+﻿using NUnit.Framework;
+using StansAssets.Foundation.Extensions;
+using UnityEngine;
+
+namespace Tests.Editor.Extensions
+{
+    public class TransformExtenstionsTest
+    {
+        [Test]
+        [TestCase(0, false, false,ExpectedResult = 0)]
```
Comment (alexey-yaremenko) [inline Tests/Editor/Extensions/TransformExtenstionsTest.cs:10]: Please, consider adding test names. So we can see great names in the _Test Runner_ window.
```suggestion
        [TestCase(0, false, false,ExpectedResult = 0, TestName = "Some good name")]
```

**Review by alexey-yaremenko — CHANGES_REQUESTED** [review]
What do you think if we rename `TransformExtensions.Clear` method to `TransformExtensions.ClearHierarchy` to make it more self-descriptive? I think that's a good idea.
Also, this method doesn't need any return value. It feels like we better make it `void`.

In addition, please, address a few of my change requests in the `TransformExtenstionsTest` class.

File: Tests/Editor/Extensions/TransformExtenstionsTest.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        [TestCase(10, false, false, ExpectedResult = 0)]
+        [TestCase(10, true, false, ExpectedResult = 0)]
+        [TestCase(10, true, true, ExpectedResult = 10)]
+        public int ClearTest(int amount, bool disactivateAtSpawn, bool deleteActiveOnly)
```
Comment (alexey-yaremenko) [inline Tests/Editor/Extensions/TransformExtenstionsTest.cs:14]:
```suggestion
        public int ClearTest(int amount, bool deactivateChildrenOnSpawn, bool deleteActiveOnly)
```

**Review by alexey-yaremenko — CHANGES_REQUESTED** [review]

File: Tests/Editor/Extensions/TransformExtenstionsTest.cs (hunk 1 above)
Comment (roman-effrimov) [reply Tests/Editor/Extensions/TransformExtenstionsTest.cs:10]: Didn't noticed that option with [TestCase]. Thanks

**Review by alexey-yaremenko — APPROVED** [review]
Simply, lovely!

**Pull request merged**